contrib/CMakeLists.txt

```diff
@@ -1,38 +1,143 @@
+
+if (CH_ODBC_ENABLE_SSL AND NOT MSVC)
+    if (CH_ODBC_PREFER_BUNDLED_SSL OR NOT OPENSSL_FOUND)
+        if (NOT OPENSSL_FOUND AND NOT CH_ODBC_PREFER_BUNDLED_SSL)
+            message (WARNING "SSL: unable to find system OpenSSL, falling back to using the bundled variant of SSL library")
+        endif ()
+
+        set (USE_SHARED 0)
 
-        # Fix iodbc include bug
-        if (NOT MSVC)
-            set_target_properties(nanodbc PROPERTIES INCLUDE_DIRECTORIES "${ODBC_INCLUDE_DIRECTORIES};${CMAKE_CURRENT_SOURCE_DIR}/nanodbc")
+        if (XCODE)
+            set (ENABLE_ASM 0 CACHE INTERNAL "")
         endif ()
-        set_target_properties(nanodbc PROPERTIES INTERFACE_LINK_LIBRARIES "${ODBC_LIBRARIES}")
+
+        add_subdirectory (ssl)
+
+        # Mimic FindOpenSSL for Poco to pickup.
+        set (OPENSSL_FOUND TRUE)
+        set (OPENSSL_INCLUDE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/ssl/include")
+        set (OPENSSL_CRYPTO_LIBRARY crypto)
+        set (OPENSSL_CRYPTO_LIBRARIES ${OPENSSL_LIBS})
+        set (OPENSSL_SSL_LIBRARY ssl)
+        set (OPENSSL_SSL_LIBRARIES ${OPENSSL_LIBS})
+        set (OPENSSL_LIBRARIES ${OPENSSL_LIBS})
     endif ()
 endif ()
 
-if (USE_INTERNAL_SSL_LIBRARY)
-    set (BUILD_SHARED 0)
-    set (USE_SHARED 0)
-    add_subdirectory (ssl)
-endif ()
+if (CH_ODBC_PREFER_BUNDLED_POCO OR NOT Poco_FOUND)
+    if (NOT Poco_FOUND AND NOT CH_ODBC_PREFER_BUNDLED_POCO)
+        message (WARNING "Poco: unable to find system Poco, falling back to using the bundled variant of the library")
+    endif ()
+
+    if (CH_ODBC_ENABLE_SSL AND MSVC)
+        set (ENABLE_NETSSL_WIN ON CACHE INTERNAL "")
+    endif ()
+
+    if (CH_ODBC_THIRD_PARTY_LINK_STATIC)
+        set (POCO_STATIC ON CACHE INTERNAL "")
+    endif ()
+
+    if (CH_ODBC_RUNTIME_LINK_STATIC)
+        set (POCO_MT ON CACHE INTERNAL "")
+    endif ()
+
+    set (CMAKE_MODULE_PATH_bak ${CMAKE_MODULE_PATH})
+    set (CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH_orig})
 
-if (USE_INTERNAL_POCO_LIBRARY)
-    set (POCO_UNBUNDLED_ZLIB 1 CACHE INTERNAL "")
     add_subdirectory (poco)
-    if (OPENSSL_FOUND AND TARGET Crypto)
-        # Bug in poco https://github.com/pocoproject/poco/pull/2100 found on macos
-        target_include_directories(Crypto PUBLIC $<BUILD_INTERFACE:${OPENSSL_INCLUDE_DIR}>)
+
+    set (CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH_bak})
+    unset (CMAKE_MODULE_PATH_bak)
+
+    if (TARGET NetSSL)
+        add_library(Poco::NetSSL ALIAS NetSSL)
     endif ()
-    if (USE_INTERNAL_POCO_LIBRARY)
-        # Remove pthread from linking poco's Net
-        set_property(TARGET Net PROPERTY INTERFACE_LINK_LIBRARIES "")
-        set_property(TARGET Foundation PROPERTY INTERFACE_LINK_LIBRARIES "")
+
+    if (TARGET NetSSLWin)
+        add_library(Poco::NetSSLWin ALIAS NetSSLWin)
     endif ()
+
+    add_library(Poco::Net ALIAS Net)
+    add_library(Poco::Foundation ALIAS Foundation)
+    add_library(Poco::Util ALIAS Util)
+
+    # Remove pthread from linking poco's Net
```

Comment:
why ?

Reply:
A legacy comment (and an action), that was copied as-is. I don't know either.